### PR TITLE
fix: tree component improvements

### PR DESCRIPTION
**Pull request by anxobotana**
Fixes extra re-renderings, adds logic to expand on hover, and optimizes some logic

### CODE REVIEW

File: src/components/Tree/Tree.tsx [hunk 1]
```diff
@@ -101,7 +101,12 @@ const reducer = produce((draft: TreeState, action: TreeStateAction) => {
 
         case 'SET_PROJECTION':
             {
+                if (isEqual(draft.projection, action.payload)) {
+                    return;
+                }
+
                 draft.projection = action.payload;
+                return;
```
Comment (fabriciomendonca) [inline src/components/Tree/Tree.tsx:109]: Do we need the added returns in the end? Wouldn't the `break` be enough for the `switch`?

File: src/components/Tree/Tree.tsx [hunk 2]
```diff
@@ -343,7 +352,9 @@ export const Tree = memo(
         };
 
         const handleDragMove = ({ delta }: DragMoveEvent) => {
-            setOffset(delta.x);
+            if (delta.x % 10 === 0) {
```
Comment (fabriciomendonca) [inline src/components/Tree/Tree.tsx:355]: Why 10? Maybe extracting it into a constant would explain better the magic number, wouldn't it?

**Review by fabriciomendonca — APPROVED** [review]

File: src/components/Tree/Tree.tsx (hunk 1 above)
Comment (anxobotana) [reply src/components/Tree/Tree.tsx:109]: no we don't need the `return` it's just a formality, _the immer way_ (https://immerjs.github.io/immer/return)

File: src/components/Tree/Tree.tsx (hunk 2 above)
Comment (anxobotana) [reply src/components/Tree/Tree.tsx:355]: yes, or making it relative to the tree padding. The point is reducing the times we refresh the offset which was triggering useless re-renderings

Comment (anxobotana) [reply src/components/Tree/Tree.tsx]: updated!

**Review by fabriciomendonca — APPROVED** [review]

**Pull request merged**